src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -380,7 +380,14 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
                     return false;
                 }
 
-                elementCount = TryGetRawDataForArrayInit(ac.InitializerOpt, out data);
+                if (ac.InitializerOpt == null)
+                {
+                    elementCount = TryGetRawDataForDefaultArray(ac, out data);
```

Comment:
Should I implement here a size limit for the field as @canton7 in #33088? E.g. generate the field only is the array smaller than 100 bytes or so.